### PR TITLE
Fixes #21877: Hardening guide for Rudder

### PR DESCRIPTION
**Pull request by amousset**
https://issues.rudder.io/issues/21877

### CODE REVIEW

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
PR updated with a new commit

File: src/reference/modules/administration/pages/security.adoc [hunk 1]
```diff
@@ -1,178 +1,126 @@
...
-
-This protocol is used for:
-
-* Policy download on Linux and AIX nodes
-* Remote run from policy servers to nodes
+=== HTTPS
 
-It uses the key pairs in `/var/rudder/cfengine-community/ppkeys/` to authenticate the node and server during
-communication.
+* By default, all HTTPS traffic is handled by the same virtual host in Apache httpd. To allow hardening the connection security, a first step is to split public Web/API from internal node-server communication. You can do so by commenting the default virtual host in `/etc/apache2/sites-enabled/rudder.conf` or `/etc/httpd/conf.d/rudder.conf` (depending on the distribution), and uncommenting the two separate configurations. You only need a way to make a difference in the configuration, in general with a different port (which allows applying different firewall rules too) or a specific domain for Web/API.
```
Comment (ncharles) [inline src/reference/modules/administration/pages/security.adoc:10]: You only need a way to make a difference in the configuration, in general with a different port (which allows applying different firewall rules too) or a specific domain for Web/API.
->
You need to define a method to distinguish both configurations, in general with a different port (which allows applying different firewall rules too) or a specific domain for Web/API.

File: src/reference/modules/administration/pages/security.adoc [hunk 2]
```diff
@@ -1,178 +1,126 @@
...
-==== Agent authentication
+[source,ini]
+----
+rudder.server.hsts=true
+rudder.server.hstsIncludeSubDomains=true
+----
 
-The first inventory sent by the node contains it's public key (apart of its certificates). Once accepted,
-it's policies are only available to a node having the matching private key.
+* You may want to hide the Apache httpd version from the headers. It cannot be done inside Rudder's configuration as it is a global parameter. To do so, you need to set the `ServerTokens` parameter to the `Prod` value (in `/etc/httpd` or `/etc/apache` depending on your distribution).
```
Comment (ncharles) [inline src/reference/modules/administration/pages/security.adoc:22]: it is a global parameter. ->  it is a global Apache2 parameter.

File: src/reference/modules/administration/pages/security.adoc [hunk 3]
```diff
@@ -1,178 +1,126 @@
...
-
-----
+To automate node acceptation on the server while checking
+the node's identity, you can use the https://docs.rudder.io/api/#tag/Nodes/operation/createNodes[node creation] API.
+You can make a call to this API in your node provisioning process, and provide
+the `agentKey` parameter (which should contain the content of `/opt/rudder/etc/ssl/agent.cert` on the node), and the `accepted` status.
 
-Now you can set your policy server with `rudder agent policy-server mypolicyserver`
-and the agent will only accept the connection if it matches the provided key hash.
+This way, once the node sends its first inventory, if it matched the node id and certificate, it will be automatically accepted. Note that this creation call also allows pre-setting node properties.
```
Comment (ncharles) [inline src/reference/modules/administration/pages/security.adoc:126]: this is misleading - the node is already existing within Rudder

This way, once the node sends its first inventory, if the node id and certificate match the pre-provisionned entry, the node inventory will be update.
Please note that you can also pre-defined node properties using this API, to ensure that the pre-provisionned node is already in the correct groups

**Review by ncharles — CHANGES_REQUESTED** [review]
This is a really neat document
Some minor changes on wording are requested to make it more understandable

**Comment by amousset** [comment]
PR updated with a new commit

**Pull request merged**